File: 819-most-common-word/819-most-common-word.js
/**
 * @param {string} paragraph
 * @param {string[]} banned
 * @return {string}
 */
var mostCommonWord = function(paragraph, banned) {
    const modParagraph = paragraph.replace(/[!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~]/g, " ")
        .toLowerCase()
        .split(" ");
    let map = {};
    for (let i = 0; i < modParagraph.length; i++){
        if (banned.includes(modParagraph[i]) || modParagraph[i] === ""){
            continue;
        } else if (modParagraph[i] in map){
            map[modParagraph[i]] += 1;
        } else {
            map[modParagraph[i]] = 1;
        }
    }
    let result = "";
    let max = 0;
    for (let word in map){
        if (map[word] > max){
            max = map[word];
            result = word;
        }
    }
    return result;
};

File: 819-most-common-word/819-most-common-word.py
class Solution:
    def mostCommonWord(self, paragraph: str, banned: List[str]) -> str:
        # replace punctuation with spaces and make all letters lower case
        normalized_s = "".join([c.lower() if c.isalnum() else " " for c in paragraph])
        # split string into words
        words = normalized_s.split()
        # create dictionary
        word_counts = defaultdict(int)
        # create set of banned words
        banned_words = set(banned)
        # count appearance of words and exclude banned words
        for word in words:
            if word not in banned_words:
                word_counts[word] += 1
        # return highest frequency word
        return max(word_counts.items(), key=lambda x: x[1])[0]